lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -121,6 +122,24 @@ struct VarExpr : public ExprBase<VarExpr, Expr::Kind::Var> {
   Expr *constraint = nullptr;
 };
 
+/// An identity expression, used to annotate the location where a variable is
+/// constrained for the first time.
```

Comment:
Could you include some more info on how this will be used later? I'm not grasping exactly what this is doing.

Reply:
Good point, that wasn't clear at all. Added a comment.